testsuite/features/support/http_client.rb

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+require 'faraday'
+
+# Wrapper class for HTTP client library (Faraday)
+class HttpClient
+  def initialize(host)
+    puts 'Activating HTTP API'
+    @http_client = Faraday.new('https://' + host)
+  end
+
+  def prepare_call(name, params)
+    short_name = name.split('.')[-1]
+    call_type =
+      if short_name.start_with?('list', 'get', 'is', 'find') || ['logout', 'errata.applicableToChannels'].include?(name)
+        'GET'
+      else
+        'POST'
+      end
+    url = '/rhn/manager/api/' + name.tr('.', '/')
+    if call_type == 'GET'
+      url += '?'
+      unless params.nil?
+        params.each do |key, value|
+          url += '&' unless url[-1] == '?'
+          url += "#{key}=#{value}"
+        end
+      end
+    end
+    [call_type, url]
+  end
+
+  def call(name, params)
+    name.sub!('auth.', '') if ['auth.login', 'auth.logout'].include?(name)
+
+    # Get session cookie from previous calls
+    if params.nil?
+      session_cookie = nil
+    else
+      session_cookie = params[:sessionKey]
```

Comment:
Why sometimes we use string keys and sometimes we use Ruby symbols? :thinking:

Reply:
rubocop has preferences about that